paimon-core/src/main/java/org/apache/paimon/table/system/SourceTableLineageTable.java

```diff
@@ -0,0 +1,190 @@
+import org.apache.paimon.types.TimestampType;
+import org.apache.paimon.types.VarCharType;
+import org.apache.paimon.utils.IteratorRecordReader;
+import org.apache.paimon.utils.ProjectedRow;
+
+import org.apache.paimon.shade.guava30.com.google.common.collect.Iterators;
+
+import javax.annotation.Nullable;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import static org.apache.paimon.utils.Preconditions.checkNotNull;
+
+/**
+ * This is a system table to display all the source table lineages.
+ *
+ * <pre>
+ *  For example:
+ *     If we select * from sys.source_table_lineage, we will get
+ *     database_name       table_name       job_name      create_time
+ *        default            test0            job1    2023-10-22 20:35:12
+ *       database1           test1            job1    2023-10-28 21:35:52
+ *          ...               ...             ...             ...
+ *     We can write sql to fetch the information we need.
+ * </pre>
+ */
+public class SourceTableLineageTable implements ReadonlyTable {
+
+    public static final String SOURCE_TABLE_LINEAGE = "source_table_lineage";
+
+    private final LineageMetaFactory lineageMetaFactory;
+    private final Map<String, String> options;
+
+    public SourceTableLineageTable(
+            LineageMetaFactory lineageMetaFactory, Map<String, String> options) {
+        this.lineageMetaFactory = lineageMetaFactory;
+        this.options = options;
+    }
+
+    @Override
+    public InnerTableScan newScan() {
+        return new ReadOnceTableScan() {
+            @Override
+            public InnerTableScan withFilter(Predicate predicate) {
+                return this;
+            }
+
+            @Override
+            protected Plan innerPlan() {
+                /// TODO get the real row count for plan.
+                return () -> Collections.singletonList((Split) () -> 1L);
+            }
+        };
+    }
+
+    @Override
+    public InnerTableRead newRead() {
+        return new SourceTableLineageRead(lineageMetaFactory, options);
+    }
+
+    @Override
+    public String name() {
+        return SOURCE_TABLE_LINEAGE;
+    }
+
+    @Override
+    public RowType rowType() {
+        List<DataField> fields = new ArrayList<>();
+        fields.add(new DataField(0, "database_name", new VarCharType(VarCharType.MAX_LENGTH)));
+        fields.add(new DataField(1, "table_name", new VarCharType(VarCharType.MAX_LENGTH)));
+        fields.add(new DataField(2, "job_name", new VarCharType(VarCharType.MAX_LENGTH)));
+        fields.add(new DataField(3, "create_time", new TimestampType()));
+        return new RowType(fields);
+    }
+
+    @Override
+    public List<String> primaryKeys() {
+        return Arrays.asList("database_name", "table_name", "job_name");
+    }
+
+    @Override
+    public Table copy(Map<String, String> dynamicOptions) {
```

Comment:
The dynamicOptions are ignored here, is it as expected?

Reply:
It's useless for lineage tables.